### DISCUSSION TITLE
Lab 3.01 Task2

### DISCUSSION THREAD

**Discussion by chape8** [rramosp/ai4eng.v1#207] [03 - GENERACION DE MODELOS]
Tengo un problema que esta tarea dado que cuando la mando a calificar me dice que la respuesta no es un numpy array pero realmente no entiendo si ese en realidad es el problema dado que ya he intentado de mil maneras y no me sigue sin dar
Este es el error que me pasa
![Error](https://user-images.githubusercontent.com/98787315/197396848-454c5213-80b5-49a3-98a5-39f4f6dc4f2d.png)

Este es mi codigo
![Lab301_Task2](https://user-images.githubusercontent.com/98787315/197396856-1c39c105-7b05-429d-b7cb-f0ddab2b3643.png)

Agradezco una mano con este inconveniente.

**Comment by Felipe-RA** [comment]
Saludos chape8

Prueba a convertir tu respuesta a un numpy array, una vez estes seguro de que estás enviando eso podemos pasar a ver qué otros posibles errores tienes.

Hasta pronto, quedo pendiente

**Reply by chape8** [reply]
Agregué la linea p=np.array(p) y puse a imprimir la dimensión de p, esto es lo que me muestra una vez la llamo

![llamado](https://user-images.githubusercontent.com/98787315/197610086-1e35a76c-1056-46ed-81ef-628a613125a1.png)

Y este es el error que me aparece ahora

![Error](https://user-images.githubusercontent.com/98787315/197610149-c4d3386a-0995-4e4f-979f-32199d432520.png)

**Reply by Felipe-RA** [reply]
Ponme aquí un enlace a una copia de tu notebook que esté abierta para echarle un ojo más directamente.

**Reply by chape8** [reply]
https://colab.research.google.com/drive/1HyhUSzq4bGsmdVbGvKb8y2pifY6mo4KJ?usp=sharing

**Reply by chape8** [reply]
Saludos Felipe-RA, si me pudiste hacerme el favor de darle una mirada al notebook que me pediste que te compartiera?

**Reply by imEag** [reply]
Hola @chape8, estuve revisando y en el notebook que compartiste el tipo de dato devuelto por la función `fit` es el correcto (numpy.ndarray) pero veo que el problema está en la lógica interna de la función, ya que, por ejemplo al ejecutar la linea `bt = fit(bX, by)` con diferentes datos hay ocasiones en las que se devuelve una respuesta vacía, cosa que no debería ocurrir, entonces cuando el calificador automático ejecuta las 100 llamadas con datos distintos, obtiene algunos resultados con longitud 0 y por eso dice que debe ser de len 2.

Además de revisar que la lógica este bien, te recomiendo comentar el código cuando las líneas sean difíciles de entender y que evites nombrar funciones solo con una letra y en vez de eso utilices nombres descriptivos y mejor en inglés.

**Reply by imEag** [reply]
Con comentar el código me refiero a poner una descripción corta y clara de lo que hace dicho segmento
